rllib/evaluation/trajectory.py

```diff
@@ -0,0 +1,274 @@
+import logging
+import numpy as np
+from typing import Dict, Optional
+
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.framework import try_import_tf, try_import_torch
+from ray.rllib.utils.numpy import convert_to_numpy
+from ray.rllib.utils.types import AgentID, EnvID, PolicyID, TensorType
+
+tf1, tf, tfv = try_import_tf()
+torch, _ = try_import_torch()
+
+logger = logging.getLogger(__name__)
+
+
+class Trajectory:
```

Comment:
Can we have unit tests for `Trajectory` class covering the edge cases here?

Reply:
sure ...

Reply:
This was added. Note that there will be more performance optimizations coming up over the next PRs. Will add a performance test (new sampling/trajectory view API vs current implementation(s)) as well.